fdbclient/FDBTypes.h

```diff
@@ -1126,4 +1126,27 @@ inline const char* transactionPriorityToString(TransactionPriority priority, boo
 	throw internal_error();
 }
 
+struct VersionVector {
+	std::map<Tag, Version> versions;
+
+	VersionVector() {}
+
+	void setVersion(const Tag& tag, Version version) {
+		versions[tag] = version;
+	}
+
+	bool hasVersion(const Tag& tag) const {
+		return versions.find(tag) != versions.end();
+	}
+
+	Version getVersion(const Tag& tag) const {
+		return versions.at(tag);
```

Comment:
This will throw an `std::out_of_range` which is not great as we don't support using standard exceptions (sadly).

I would suggest to change the code like this:
```suggestion
		auto iter = versions.find(tag);
		ASSERT(iter != versions.end());
		return *iter;
```

It's sadly less readable than your code, but I think this will do what you want it to do. Additionally you probably should add a comment to the method stating that the tag existing in the version vector is a precondition